### PR TITLE
Fix client go registration for mixer types

### PR DESCRIPTION
**Pull request by jasonwzm**
client-go fix for https://github.com/istio/istio/issues/19638

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,14 +9,12 @@ require (
...
 	golang.org/x/time v0.0.0-20190921001708-c4c64cad1fd0 // indirect
 	google.golang.org/appengine v1.6.5 // indirect
 	google.golang.org/genproto v0.0.0-20190916214212-f660b8655731 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
-	gopkg.in/inf.v0 v0.9.1 // indirect
-	istio.io/api v0.0.0-20191109011911-e51134872853
-	k8s.io/apimachinery v0.0.0-20191004115801-a2eda9f80ab8
+	istio.io/api v0.0.0-20200106191429-c785926a849c
+	k8s.io/apimachinery v0.17.0
 	k8s.io/client-go v0.0.0-20191016111102-bec269661e48
```
Comment (howardjohn) [inline go.mod]: use v0.17.0?

Comment (jasonwzm) [reply go.mod]: k8s client-go changed the versioning name for their repos: https://github.com/kubernetes/client-go#client-go

Comment (howardjohn) [reply go.mod]: istio/istio uses v0.17.0. I think the way you have it now is the old way

Comment (jasonwzm) [reply go.mod]: k8s.io/client-go is modified to use the new way.

Comment (howardjohn) [reply go.mod]: nice, thanks! that will make consumption much easier probably

**Pull request merged**